### PR TITLE
Sp 06 mon

### PR DESCRIPTION
**Pull request by fyarkiy**

### CODE REVIEW

File: src/main/java/com/cinema/controllers/CinemaHallController.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    @PostMapping
+    public CinemaHall addCinemaHall(@RequestBody CinemaHallRequestDto cinemaHallRequestDto) {
+        return cinemaHallService.add(cinemaHallMapper.mapDtoToCinemaHall(cinemaHallRequestDto));
+    }
+
+    @GetMapping
+    public List<CinemaHallResponseDto> getAll() {
+        return cinemaHallService.getAll().stream()
+                .map(ch -> cinemaHallMapper.mapCinemaHallToDto(ch))
```
Comment (Borisko130) [inline src/main/java/com/cinema/controllers/CinemaHallController.java]: You can simplify this to
````cinemaHallMapper::mapCinemaHallToDto````

**Review by Borisko130 — APPROVED** [review]

File: src/main/java/com/cinema/controllers/CinemaHallController.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/cinema-halls")
+public class CinemaHallController {
+    private CinemaHallService cinemaHallService;
```
Comment (Rudiachenko) [inline src/main/java/com/cinema/controllers/CinemaHallController.java]: let's make it final

File: src/main/java/com/cinema/controllers/ShoppingCartController.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        shoppingCartService.addSession(movieSessionService.getById(movieSessionId),
+                userService.getById(userId));
+    }
+
+    @GetMapping("/by-user/{userId}")
+    public ShoppingCartResponseDto getByUser(@PathVariable Long userId) {
+        return shoppingCartMapper
+                .mapCartToDto(shoppingCartService.getByUser(userService.getById(userId)));
+    }
+
```
Comment (Rudiachenko) [inline src/main/java/com/cinema/controllers/ShoppingCartController.java]: redundant line

File: src/main/java/com/cinema/service/impl/mapper/CinemaHallMapper.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import com.cinema.model.CinemaHall;
+import com.cinema.model.dto.CinemaHallRequestDto;
+import com.cinema.model.dto.CinemaHallResponseDto;
+import org.springframework.stereotype.Component;
+
+@Component
+public class CinemaHallMapper {
+
+    public CinemaHall mapDtoToCinemaHall(CinemaHallRequestDto cinemaHallRequestDto) {
+        return new CinemaHall(cinemaHallRequestDto.getCapacity(),
```
Comment (Rudiachenko) [inline src/main/java/com/cinema/service/impl/mapper/CinemaHallMapper.java:12]: "In your mappers prefer setters over big constructors while converting entity to dto and vice versa." same in other cases

**Review by Rudiachenko — CHANGES_REQUESTED** [review]

File: src/main/java/com/cinema/controllers/CinemaHallController.java (hunk 1 above)
Comment (fyarkiy) [reply src/main/java/com/cinema/controllers/CinemaHallController.java]: Thanks! Will do then everywhere

File: src/main/java/com/cinema/controllers/CinemaHallController.java (hunk 2 above)
Comment (dmytro-hryhoruk) [reply src/main/java/com/cinema/controllers/CinemaHallController.java]: It's not crucial.This Bean is singleton anyway and trying to reassign this variable like this somewhere in your code:
`cinemaHallService = new CinemaHallServiceIml(); `
You'll mostly end up with an exception.

File: src/main/java/com/cinema/controllers/OrderController.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    @PostMapping("/complete")
+    public Order completeOrder(@RequestParam Long userId) {
+        ShoppingCart shoppingCart = shoppingCartService.getByUser(userService.getById(userId));
+        return ordersService.completeOrder(shoppingCart.getTickets(), shoppingCart.getUser());
+    }
+
+    @GetMapping
+    public List<OrderResponseDto> orders(@RequestParam Long userId) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/cinema/controllers/OrderController.java]: is that a fine naming for this method?

File: src/main/java/com/cinema/service/impl/mapper/CinemaHallMapper.java (hunk 4 above)
Comment (fyarkiy) [reply src/main/java/com/cinema/service/impl/mapper/CinemaHallMapper.java:12]: Quite judgemental - what is big and what isn't. I thought this case is not big... Let's see what mentor says

File: src/main/java/com/cinema/controllers/ShoppingCartController.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    @PostMapping("/movie-sessions")
+    public void addMovies(@RequestParam Long movieSessionId,
+                          @RequestParam Long userId) {
+        shoppingCartService.addSession(movieSessionService.getById(movieSessionId),
+                userService.getById(userId));
+    }
+
+    @GetMapping("/by-user/{userId}")
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/cinema/controllers/ShoppingCartController.java]: user id it's a temporary variable, let's not make it a part of url

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**